airbyte-integrations/connectors/source-rd-station-marketing/source_rd_station_marketing/streams.py

```diff
@@ -0,0 +1,203 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+from abc import ABC
+from datetime import date
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class RDStationMarketingStream(HttpStream, ABC):
+    url_base = "https://api.rd.services"
+    primary_key = None
+    page = 2
+    page_size=125
+    extra_params = {}
+    data_field = None
+
+    def __init__(self, authenticator, replication_start_date=None, **kwargs):
+        super().__init__(authenticator=authenticator, **kwargs)
+        self._replication_start_date = replication_start_date
+
+    def path(self, **kwargs) -> str:
+        class_name = self.__class__.__name__
+        return f"/platform/{class_name[0].lower()}{class_name[1:]}"
+    
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if self.data_field:
+            json_response = response.json().get(self.data_field)
+        else:
+            json_response = response.json()
+        if json_response:
+            page_params = dict(page=self.page)
+            self.page = self.page + 1
+            return page_params
+        else:
+            return None
```

Comment:
There is any situation where you don't have the data field value?
Also being more defensive would be better a try/catch and fail if something doesn't work as expected.

I think other connectors with page value use a better approach:
```python
json_response = response.json()
count_records = count(json_response.get("items"))
if count_records <= PAGE_SIZE_LIMIT:
    return None
else: 
    self.page += 1
    return {"page": self.page}
```

If not clear if RD Station return any metadata in the API response. Would be nice if you can share one example of it.

Reply:
Some endpoints respond with the number of total rows/pages on the header of the response, but others don't :( Also, I got a case where the number wasn't correct, so I ditched the use of this metadata altogether

Reply:
> There is any situation where you don't have the data field value?
Yes! Popups, landing pages and embeddables respond with a list, that's why the if/else in some parts

Reply:
I'll mark this conversation as resolved, but let me know if you have any questions!